02-Challenge/Develop/script.js

```diff
@@ -11,30 +11,44 @@ var generateBtn = document.querySelector("#generate");
 
 
 // Write password to the #password input
-function writePassword() {
-   var correctPrompt = getPrompt(); 
-  var passwordText = document.querySelector("#password");
-
-    if(correctPrompt) {
-      var newPassword = generatePassword();
-      passwordText.value = newPassword;
+// function writePassword() {
+//    var correctPrompt = getPrompt(); 
+//   var passwordText = document.querySelector("#password");
+
+//     if(correctPrompt) {
+//       var newPassword = generatePassword();<-- You wrote a function to collect an array based on user choices (getPrompt).
+// Then you wrote a function to loop through that array and select items (generatePassword).
+// You want to generatePassword based on your getPrompt, so you have to pass getPrompt into your generatePassword fuction so that generatePassword can start working with that data.
+// If you don't pass anything into it, generatePassword doesn't have any data to work with.  See line 33.
+//       passwordText.value = newPassword;
    
-    }  else {
-      passwordText.value = "";
-    } 
+//     }  else {
+//       passwordText.value = "";
+//     } 
       
+// }
+
+function writePassword() {
+  var correctPrompt = getPrompt();
+  var newPassword = generatePassword(correctPrompt);
+  var passwordText = document.querySelector("#password");
+  passwordText.value = newPassword;
 }
 
 function generatePassword() {
-    var password = "";
```

Comment:
I played around with this, but ended up copying and pasting your original code back, so I didn't actually change anything here.